app/registries/foreman/plugin.rb

```diff
@@ -53,6 +48,24 @@ class << self
         :graphql_types_registry, :fact_importer_registry
       private :new
 
+      def initialize_registries(registries = {})
+        @fact_importer_registry = registries[:fact_importer_registry] || Plugin::FactImporterRegistry.new
+        @report_scanner_registry = registries[:report_scanner_registry] || Plugin::ReportScannerRegistry.new
+        @report_origin_registry = registries[:report_origin_registry] || Plugin::ReportOriginRegistry.new
+        @medium_providers = registries[:medium_providers] || Plugin::MediumProvidersRegistry.new
+        @graphql_types_registry = registries[:graphql_types_registry] || Plugin::GraphqlTypesRegistry.new
+      end
+
+      def registries
+        {
+          fact_importer_registry: @fact_importer_registry,
+          report_scanner_registry: @report_scanner_registry,
+          report_origin_registry: @report_origin_registry,
+          medium_providers: @medium_providers,
+          graphql_types_registry: @graphql_types_registry,
+        }
+      end
```

Comment:
Does it make sense to use `@registries` with `attr_reader :registeries` instead? Perhaps even a DSL method `def registry(name, registry)` to declare a registry. That way plugins can add new registries.

Reply:
Make sense to me :+1: